doc/BGP/BGP-supress-fib-pending.md

```diff
@@ -0,0 +1,862 @@
+
+```jinja2
+router bgp {{ DEVICE_METADATA['localhost']['bgp_asn'] }}
+{% if DEVICE_METADATA['localhost']['bgp-suppress-fib-pending'] == 'enabled' %}
+  bgp suppress-fib-pending
+{% endif %}
+```
+
+Zebra shall be started with an option ```--asic-offload=notify_on_offload```. This option will make zebra wait for ```RTM_F_OFFLOAD``` before notifying about installation status to bgpd. The template *supervisor.conf.j2* file is updated:
+
+```jinja2
+[program:zebra]
+{% if DEVICE_METADATA['localhost']['bgp-suppress-fib-pending'] == 'enabled' %}
+{% set zebra_extra_arguments = "--asic-offload=notify_on_offload" }
+{% endif %}
+command=/usr/lib/frr/zebra -A 127.0.0.1 -s 90000000 -M dplane_fpm_nl {{ zebra_extra_arguments }}
+```
+
+Before the route is offloaded ```show ip route``` shows routes as queued:
+
+```
+admin@sonic:~$ show ip route
+Codes: K - kernel route, C - connected, S - static, R - RIP,
+       O - OSPF, I - IS-IS, B - BGP, E - EIGRP, N - NHRP,
+       T - Table, v - VNC, V - VNC-Direct, A - Babel, F - PBR,
+       f - OpenFabric,
+       > - selected route, * - FIB route, q - queued, r - rejected, b - backup
+       t - trapped, o - offload failure
+
+B>q 0.0.0.0/0 [20/0] via 10.0.0.1, PortChannel102, weight 1, 00:04:46
+  q                  via 10.0.0.5, PortChannel105, weight 1, 00:04:46
+  q                  via 10.0.0.9, PortChannel108, weight 1, 00:04:46
+  q                  via 10.0.0.13, PortChannel1011, weight 1, 00:04:46
+```
+
+Once zebra is notified about successfull route offload status the *offload* flag is set:
+
+```
+admin@sonic:~$ vtysh -c "show ip route 100.1.0.25/32 json"
+{
+    "100.1.0.25/32": [
+        {
+            "prefix": "100.1.0.25/32",
+            "prefixLen": 32,
+            "protocol": "bgp",
+            "vrfId": 0,
+            "vrfName": "default",
+            "selected": true,
+            "destSelected": true,
+            "distance": 20,
+            "metric": 0,
+            "installed": true,
+            "offloaded": true,
+            "table": 254,
+            "internalStatus": 80,
+            "internalFlags": 264,
+            "internalNextHopNum": 1,
+            "internalNextHopActiveNum": 1,
+            "nexthopGroupId": 4890,
+            "installedNexthopGroupId": 4890,
+            "uptime": "00:22:20",
+            "nexthops": [
+                {
+                    "flags": 3,
+                    "fib": true,
+                    "ip": "10.0.0.49",
+                    "afi": "ipv4",
+                    "interfaceIndex": 788,
+                    "interfaceName": "PortChannel1013",
+                    "active": true,
+                    "weight": 1
+                }
+            ]
+        }
+    ]
+}
+```
+
+#### 7.3. RouteOrch
+
+<!-- omit in toc -->
+#### ResponsePublisher in RouteOrch
+
+```RouteOrch``` need to use existing ```ResponsePublisher``` API to be able to publish responses into ```APPL_STATE_DB```.

```

Comment:
@qiluo-msft , could you please review this section?

Reply:
Comment added.